### PR TITLE
socket_base: remove deprecated header files

### PR DESCRIPTION
**Pull request by authmillenon**
Removes (hopefully) all very last remnants of the old network stack

Depends on ~~#3646~~ (merged), ~~#3876~~ (merged), ~~#3878~~ (merged) and ~~#3882~~ (merged).

### CODE REVIEW

**Comment by authmillenon** [comment]
Rebased to current master and dependencies.

**Comment by authmillenon** [comment]
Rebased to current dependencies and master.

File: Makefile.dep [hunk 1]
```diff
@@ -2,6 +2,13 @@ ifneq (,$(filter libcoap,$(USEPKG)))
     USEMODULE += posix_sockets
 endif
 
+ifneq (,$(filter nhdp,$(USEMODULE)))
+  USEMODULE += conn_udp
+  USEMODULE += vtimer
+  USEMODULE += oonf_common
```
Comment (haukepetersen) [inline Makefile.dep]: no need, you state below that `oonf_rfc5444` depends on it.

File: Makefile.dep [hunk 2]
```diff
@@ -297,7 +304,6 @@ endif
 
 ifneq (,$(filter nhdp,$(USEMODULE)))
```
Comment (haukepetersen) [inline Makefile.dep]: please remove this, as you re-introduced it some lines above..

File: Makefile.dep (hunk 1 above)
Comment (authmillenon) [inline Makefile.dep]: Please review changes to NHDP and OONF in #3876 and #3882. This PR just removes the files related to `socket_base `

File: Makefile.dep (hunk 2 above)
Comment (authmillenon) [inline Makefile.dep]: Dito. This is even propably just a merge error resulting from the (temporary) merge in this PR

Comment (authmillenon) [inline Makefile.dep]: (it is, just checked it in #3876)

**Comment by authmillenon** [comment]
Fixed the merge conflict error, so that @haukepetersen is happy with the state of this PR :-)

**Comment by haukepetersen** [comment]
looks better. Will give a second look once the dependency is merged and its rebased...

**Comment by authmillenon** [comment]
Rebased to current master and #2772.

**Comment by authmillenon** [comment]
Rebased to current #2772

**Comment by authmillenon** [comment]
Rebased to current masetr

**Comment by authmillenon** [comment]
(not waiting on another PR anymore)

**Comment by haukepetersen** [comment]
ACK and go.

**Comment by authmillenon** [comment]
Yay, last PR (for now) for RC merged.

**Pull request merged**